packages/components/src/plugins/index.js

```diff
@@ -52,36 +53,36 @@ export class Plugins extends Component {
 			return false;
 		}
 
-		const installs = await Promise.all(
-			pluginSlugs.map( async ( slug ) => {
-				return await installPlugin( slug );
-			} )
-		);
-
-		const installErrors = installs.filter(
-			( install ) => install.status !== 'success'
-		);
+		const installs = await installPlugins( pluginSlugs );
 
-		if ( installErrors.length ) {
-			this.handleErrors( installErrors );
+		if ( installs.errors && Object.keys( installs.errors.errors ).length ) {
+			this.handleErrors( installs.errors.errors );
 			return;
 		}
 
 		const activations = await activatePlugins( pluginSlugs );
 
-		if ( activations.status === 'success' ) {
-			this.handleSuccess( activations.activePlugins );
+		if ( activations.success && activations.data.activated ) {
+			this.handleSuccess( activations.data.activated );
 			return;
 		}
 
-		this.handleErrors( activations );
+		this.handleErrors( activations.errors.errors );
 	}
 
 	handleErrors( errors ) {
 		const { onError, createNotice } = this.props;
 
-		errors.forEach( ( error ) => {
-			createNotice( 'error', error );
+		Object.keys( errors ).forEach( ( plugin ) => {
+			createNotice(
+				'error',
+				pluginNames[ plugin ]
+					? errors[ plugin ][ 0 ].replace(
+							`\`${ plugin }\``,
+							pluginNames[ plugin ]
+					  )
+					: errors[ plugin ][ 0 ]
```

Comment:
A comment saying you're replacing a slug with a name, if it exists, would be helpful.